tests/ha/test_server_pod_failure.py

```diff
@@ -505,3 +506,377 @@ def test_degraded_writes_safe_server_pod_shutdown(self):
+        self.restore_method = const.RESTORE_DEPLOYMENT_K8S
+        LOGGER.info(
+            "Step 3: Successfully shutdown/deleted server pod %s by deleting deployment ", pod_name)
+
+        LOGGER.info("Step 4: Check cluster status")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_false(resp[0], resp)
+        LOGGER.info(
+            "Step 4: Cluster has some failures due to server pod %s has gone down.", pod_name)
+
+        LOGGER.info("Step 5: Check services status that were running on server pod %s", pod_name)
+        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod_name], fail=True,
+                                                           hostname=hostname)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 5: Services of server pod %s are in offline state", pod_name)
+
+        pod_list.remove(pod_name)
+        LOGGER.info("Step 6: Check services status on remaining pods %s", pod_list)
+        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=pod_list, fail=False)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 6: Services status on remaining pods are in online state.")
+
+        event.clear()
+        thread.join()
+        del_resp = tuple()
+        LOGGER.info(
+            "Step 7: Verify status for In-flight DELETEs while server pod %s going down", pod_name)
+        while len(del_resp) != 2:
+            del_resp = del_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not del_resp:
+            assert_utils.assert_true(False, "No logs from background DELETEs")
+        event_del_bkt = del_resp[0]
+        fail_del_bkt = del_resp[1]
+        assert_utils.assert_false(len(fail_del_bkt),
+                                  f"Bucket DELETEs failed when no server pod down {fail_del_bkt}")
+        assert_utils.assert_true(len(event_del_bkt),
+                                 f"No bucket DELETEs failed during server pod down {event_del_bkt}")
+        LOGGER.info("Step 7: Verified status for In-flight DELETEs while server pod %s going "
+                    "down", pod_name)
+
+        LOGGER.info("Cleaning up IAM user data")
+        resp = self.ha_obj.delete_s3_acc_buckets_objects(self.s3_clean)
+        assert_utils.assert_true(resp[0], resp[1])
+        self.s3_clean.pop('s3_acc')
+
+        LOGGER.info("Step 8: Create IAM user with multiple buckets and run IOs when cluster has "
+                    "some failures due to server pod %s going down.", pod_name)
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.s3_clean.update(users)
+        self.test_prefix = 'test-39910-1'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipcleanup=True,
```

Comment:
No need to skipcleanup as we expect delete working without any issue in this test.